aws-opentelemetry-distro/src/amazon/opentelemetry/distro/patches/_botocore_patches.py

```diff
@@ -65,6 +68,38 @@ def patch_extract_attributes(self, attributes: _AttributeMapT):
     _SqsExtension.extract_attributes = patch_extract_attributes
 
 
+def _apply_botocore_sns_patch() -> None:
+    """Botocore instrumentation patch for SNS
+
+    This patch extends the existing upstream extension for SNS. Extensions allow for custom logic for adding
+    service-specific information to spans, such as attributes. Specifically, we are adding logic to add
+    "aws.sns.topic_arn" attributes to be used to generate AWS_REMOTE_RESOURCE_TYPE and AWS_REMOTE_RESOURCE_IDENTIFIER.
+    Callout that today, the upstream logic adds SpanAttributes.MESSAGING_DESTINATION_NAME,
+    but we are not using it as it can be assigned with TargetArn as well.
```

Comment:
The upstream also assign TopicArn to `SpanAttributes.MESSAGING_DESTINATION_NAME` [here](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/7bddbb54195be4004275c83ef2592b362e443dc8/instrumentation/opentelemetry-instrumentation-botocore/src/opentelemetry/instrumentation/botocore/extensions/sns.py#L82). I want to call out we are not using this attribute because it can also be TargetARN. Instead, we assign a new attribute "aws.sns.topic_arn". I can remove it if it is not needed.

Reply:
Remove it as discussed offline.